packages/functions/src/lambda/task/update/lambda.test.ts

```diff
@@ -0,0 +1,76 @@
+import { afterEach,  describe, expect, test, vi} from 'vitest';
+import { APIGatewayProxyEventV2 } from 'aws-lambda';
+import { main } from "./lambda"
+import { Task } from '@kanban-board-team-aws/functions/model/Task';
+import TaskRepository from '@kanban-board-team-aws/functions/repositories/taskRepository';
+
+const taskMock = {id: "7e76c64a-7270-4a65-8f17-fe9ac3187b1a",name: "name",description : "testDesc",columnId: 1, order: 1} as Task;
+
+describe("/task/update tests", ()=>{
+
+    afterEach(_=>{
+        vi.restoreAllMocks()
+    })
+
+    test(`should return status code 200 and "Updated task" message`, async ()=>{
+
+
+        
+        // GIVEN
+        vi.spyOn(TaskRepository.prototype, "put").mockResolvedValue()
+
+        const event: APIGatewayProxyEventV2 = {
+            body: JSON.stringify(taskMock)
+        } as any
+
+        // WHEN
+        const result = await main(event)
+
+        // THEN
+        expect(result?.statusCode ?? 0).toBe(200)
+        expect(JSON.parse(result?.body ?? "")).toBe(`Updated task.`)
+    })
+
+    test.each([
+        ["missing body fields", 
+            {a: "1"}
+        ],
+        ["incorrect body fields",
+            {
+                id: taskMock.id,
+                name: taskMock.name,
+                description: taskMock.description,
+                columnId: "this shouldn't be a string",
+                order: taskMock.order
+            }
+        ]
+    ])(`should return status code 400 - %s`, async (description, body)=>{       
+        // GIVEN
+        vi.spyOn(TaskRepository.prototype, "put").mockResolvedValue()
+
+        const event: APIGatewayProxyEventV2 = {
+            body: JSON.stringify(body)
+        } as any
+
+        //WHEN
+        const result = await main(event)
+
+        // THEN
+        expect(result?.statusCode ?? 0).toBe(400)
+    })
+
+    test("should return status code 500", async () => {
+        // GIVEN
+        vi.spyOn(TaskRepository.prototype, "put").mockRejectedValue(undefined)
```

Comment:
What is returning database on invalid request? Did you check that? Probably will reject with Error

Reply:
documentation doesn't really tell us a lot about possible errors and it is quite hard to fail requests to dynamoDB and it is already quite foolproof